### PR TITLE
fix(argo-cd): Quote annotation values

### LINKED ISSUE #1106: Argo-CD - Redis - ReadString: expects " or n, but found t
### Describe the bug

I got this error, when using the Chart with Terraforms `helm_release`, adding to the metadata via `set`.

```
  set {
    name  = "redis.service.annotations.service\\.beta\\.kubernetes\\.io/azure-load-balancer-internal"
    value = "true"
  }
```

The complete error message:
```
ReadString: expects " or n, but found t, error found in #10 byte of ...|nternal":true},"labe|..., bigger context ...|beta.kubernetes.io/azure-load-balancer-internal":true},"labels":{"app.kubernetes.io/component":"redi|...
```

The workarounds described [here](https://github.com/hashicorp/terraform-provider-helm/issues/556), works:
```
  set {
    name  = "redis.service.annotations.service\\.beta\\.kubernetes\\.io/azure-load-balancer-internal"
    value = "true"
    type  = "string"
  }
```

Also this works:
```
resource "helm_release" "argocd" {
  name       = "argo-cd"
  repository = "https://argoproj.github.io/argo-helm"
  chart      = "argo-cd"

  values = [<<EOF
redis:
  service:
    annotations:
      service.beta.kubernetes.io/azure-load-balancer-internal: "true"
EOF
  ]

}
```

But probably, this should be fixed in the service template by quoting the value, as it is already done in the other templates.
E.g. here for dex: https://github.com/argoproj/argo-helm/blob/df3b2cf96a3f3ef775f611fa22d2c80b94444f7a/charts/argo-cd/templates/dex/service.yaml#L9

### Related helm chart

argo-cd

### To Reproduce

```
resource "helm_release" "argocd" {
  name       = "argo-cd"
  repository = "https://argoproj.github.io/argo-helm"
  chart      = "argo-cd"

  set {
    name  = "redis.service.annotations.service\\.beta\\.kubernetes\\.io/azure-load-balancer-internal"
    value = "true"
  }

}
```

### Expected behavior

There should be no difference between setting the metadata via `set` or via `values`, using escaped quotes in the value, or explicitly stating the type should not be necessary.

### Screenshots

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Treehopper**
* Quote annotation values in redis
* Update chart version

Resolves #1106

---
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Comment by davidkarlsen** [comment]
@Treehopper can you add a comment here too please: https://github.com/argoproj/argo-helm/blob/master/charts/argo-cd/Chart.yaml#L24

**Review by davidkarlsen — CHANGES_REQUESTED** [review]
see inline comments

File: charts/argo-cd/templates/redis/service.yaml [hunk 1]
```diff
@@ -11,7 +11,8 @@ metadata:
     {{- end }}
   {{- with .Values.redis.service.annotations }}
   annotations:
-    {{- toYaml . | nindent 4 }}
+  {{- range $key, $value := .Values.redis.service.annotations }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/redis/service.yaml:14]: Scope ist different here as we use `with`
```suggestion
  {{- range $key, $value := . }}
```

File: charts/argo-cd/Chart.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 apiVersion: v2
-appVersion: v2.2.3
+appVersion: v2.2.4
```
Comment (mkilchhofer) [inline charts/argo-cd/Chart.yaml:2]: Wrong version. Please revert `appVersion` and bump `version`.

Also please update the changelog on line 24

**Review by mkilchhofer — CHANGES_REQUESTED** [review]
I think besides my comments, this snipped exists on various resources in the ArgoCD helm chart. We should change these as well.

File: charts/argo-cd/templates/redis/service.yaml [hunk 3]
```diff
@@ -11,7 +11,8 @@ metadata:
     {{- end }}
   {{- with .Values.redis.service.annotations }}
   annotations:
-    {{- toYaml . | nindent 4 }}
+  {{- range $key, $value := . }}
+    {{ $key }}: {{ $value | quote }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/redis/service.yaml:15]: `{{- end }}` missing
```suggestion
    {{- range $key, $value := . }}
    {{ $key }}: {{ $value | quote }}
    {{- end }}
```

**Review by mkilchhofer — CHANGES_REQUESTED** [review]

File: charts/argo-cd/templates/redis/service.yaml (hunk 3 above)
Comment (Treehopper) [reply charts/argo-cd/templates/redis/service.yaml]: Thanks for your patience. I'm new to this template language, and don't have a proper setup to test this here...

Comment (mkilchhofer) [reply charts/argo-cd/templates/redis/service.yaml]: No problem, we are happy to see contributions 🙏

**Comment by Treehopper** [comment]
> I think besides my comments, this snipped exists on various resources in the ArgoCD helm chart. We should change these as well.

In my application, I set the loadbalancer annotation on all Argo-CD services in the same way. Redis is the only one that caused the error. Though, I didn't check for similar issues elsewhere.

**Comment by mkilchhofer** [comment]
@Treehopper I looked trough all occurrences of annotations and replaced every toYaml with the existing lines to quote the annotation values.

@davidkarlsen and @mbevc1 can you also have a look into it? I will not approve the changes as I introduces some of them ;-)

**Review by mbevc1 — APPROVED** [review]
LGTM :+1:

**Pull request merged**